scripts/find_all_methods.jl

```diff
@@ -0,0 +1,148 @@
+#=
+For a given project (or package), finds all methods defined in that project
+
+Usage: `julia preprocess.jl PROJECT_DIR OUT_DIR`
+    PROJECT_DIR ... directory with the Project.toml file we want to analyze
+    OUT_DIR ....... directory where output is written, also a fresh depot is maintained here
+
+Notes:
+    - We look at the project's main module, all its transitive dependencies' toplevel modules, and the Base module
```

Comment:
Apart from `Base`, you need to special-case `Core` (not sure how often people extend stuff from `Core` but they probably do it sometimes). This is documented https://docs.julialang.org/en/v1/manual/modules/#Default-top-level-definitions-and-bare-modules

Reply:
I thought you can't extend anything from Core as it contains builtin functions?

Reply:
I see, not only..

Reply:
Right. You need to distinguish functions (names) from methods (implementations). Core can define a function name foo and say that there's a built-in implementation of foo for the input of type Bar. Later on, a user module can add another, normal (non built-in) method for the same function (name) but different input type.